irods_query.hpp

```diff
@@ -162,6 +161,23 @@ namespace irods {
         class gen_query_impl : public query_impl_base {
             public:
             virtual ~gen_query_impl() {
+                // Close all statements for this query
+                gen_input_.continueInx = gen_output_->continueInx + 1;
+                freeGenQueryOut(&gen_output_);
+                gen_input_.maxRows = 0;
+                while (gen_input_.continueInx > 0) {
+                    auto err = query_helper::gen_query_fcn(
+                                   comm_,
+                                   &gen_input_,
+                                   &gen_output_);
+                    if (err < 0) {
+                        irods::log(ERROR(err, (boost::format(
+                                    "[%s] - Failed to close statement with continueInx [%d]") %
```

Comment:
mention this is gen vs spec in the error message?

Reply:
Done

Reply:
Actually, we don't need to do this since we are printing out the __FUNCTION__. So the message will say ~spec_query_impl or ~gen_query_impl at the beginning.

Reply:
good good